### ISSUE TITLE
AMP metrics ingest for running in EKS on Fargate fails

### ISSUE THREAD

**Issue by mhausenblas** [aws-observability/aws-otel-collector#542]
TL;DR: When I run the ADOT collector in EKS on Fargate, the metrics pipeline does not seem to ingest data into AMP.

I have the following setup: a EKS on Fargate cluster and a Go application that exposes OpenMetrics. When I use an in-cluster Prometheus agent, I can `remote_write` into AMP but with the same config, the metrics pipeline of the ADOT collector does not seem to work for me, while the traces pipeline work just fine (ingesting data into X-Ray). AFAICT the error seems to be an `invalid temporality and type combination` (see details in the collector logs, below)

## Setup

EKS config (applied with `eksctl`):

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: test
  region: eu-west-1
  version: "1.19"
iam:
  withOIDC: true
  serviceAccounts:
  - metadata:
      name: adot-collector
      namespace: ho11y
      labels: {aws-usage: "application"}
    attachPolicy:
      Version: "2012-10-17"
      Statement:
      - Effect: Allow
        Action:
        - "aps:RemoteWrite"
        - "aps:GetSeries"
        - "aps:GetLabels"
        - "aps:GetMetricMetadata"
        - "aps:QueryMetrics"
        - "logs:PutLogEvents"
        - "logs:CreateLogGroup"
        - "logs:CreateLogStream"
        - "logs:DescribeLogStreams"
        - "logs:DescribeLogGroups"
        - "xray:PutTraceSegments"
        - "xray:PutTelemetryRecord"
        - "xray:GetSamplingRules"
        - "xray:GetSamplingTargets"
        - "xray:GetSamplingStatisticSummaries"
        - "ssm:GetParameters"
        Resource: "*"
fargateProfiles:
- name: defaultfp
  selectors:
  - namespace: ho11y
  - namespace: kube-system
cloudWatch:
  clusterLogging:
    enableTypes: ["*"]
```

ADOT collector config:

```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: adot-collector
  labels:
    app: aws-adot
    component: adot-collector
data:
  adot-collector-config: |
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: 0.0.0.0:4317
          http:
            endpoint: 0.0.0.0:55681
      awsxray:
        endpoint: 0.0.0.0:2000
        transport: udp
      prometheus:
        config:
          global:
            scrape_interval: 15s
            scrape_timeout: 10s
          scrape_configs:
          - job_name: frontend
            kubernetes_sd_configs:
            - role: pod
            tls_config:
              ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
              insecure_skip_verify: true
            bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
            relabel_configs:
            - source_labels: [__meta_kubernetes_namespace]
              action: replace
              target_label: kubernetes_namespace
            - source_labels: [__meta_kubernetes_pod_name]
              action: replace
              target_label: kubernetes_pod_name
            - source_labels: [__meta_kubernetes_pod_label_app]
              action: keep
              regex: frontend
    exporters:
      awsxray:
      awsemf:
      awsprometheusremotewrite:
        endpoint: "https://aps-workspaces.eu-west-1.amazonaws.com/workspaces/ws-xxxxxxxxxxxxx/api/v1/remote_write"
        aws_auth:
          region: "eu-west-1"
          service: "aps"
        namespace: "ho11y"
      logging:
        loglevel: debug
    extensions:
      health_check:
      pprof:
      zpages:
    service:
      extensions: [pprof, zpages, health_check]
      pipelines:
        traces:
          receivers: [otlp, awsxray]
          # processors: [batch/traces]
          exporters: [awsxray]
        metrics:
          receivers: [prometheus]
          # processors: [batch/metrics]
          exporters: [logging, awsprometheusremotewrite]

---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: adot-collector
rules:
  - apiGroups: [""]
    resources:
    - nodes
    - nodes/proxy
    - services
    - endpoints
    - pods
    verbs: ["get", "list", "watch"]
  - apiGroups:
    - extensions
    resources:
    - ingresses
    verbs: ["get", "list", "watch"]
  - nonResourceURLs: ["/metrics"]
    verbs: ["get"]
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: adot-collector-binding
subjects:
  - kind: ServiceAccount
    name: adot-collector
    namespace: ho11y
roleRef:
  kind: ClusterRole
  name: adot-collector
  apiGroup: rbac.authorization.k8s.io
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: adot
  labels:
    app: aws-adot
    component: adot-collector
spec:
  selector:
    matchLabels:
      app: adot
  replicas: 1
  template:
    metadata:
      labels:
        app: adot
    spec:
      serviceAccountName: adot-collector
      containers:
      - name: adot-collector
        image: public.ecr.aws/aws-observability/aws-otel-collector:v0.10.0
        command:
        - "/awscollector"
        - "--config=/conf/adot-collector-config.yaml"
        - "--log-level=DEBUG"
        ports:
        - containerPort: 55681
        - containerPort: 55680
        - containerPort: 4317
        - containerPort: 8888
        env:
        - name: AWS_REGION
          value: "eu-west-1"
        volumeMounts:
        - name: adot-collector-config-vol
          mountPath: /conf
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        readinessProbe:
          httpGet:
            path: /
            port: 13133
      volumes:
        - configMap:
            name: adot-collector
            items:
              - key: adot-collector-config
                path: adot-collector-config.yaml
          name: adot-collector-config-vol
---
apiVersion: v1
kind: Service
metadata:
  name: adot
  labels:
    app: aws-adot
    component: adot-collector
spec:
  ports:
  - name: receiver-http
    port: 55681
  - name: receiver-grpc-legacy
    port: 55680
  - name: receiver-grpc
    port: 4317
  - name: metrics
    port: 8888
  selector:
    app: adot
```

## Collector logs

I see the following in the ADOT logs:

```
2021/06/13 11:04:36 AWS OTel Collector version: v0.10.0
2021/06/13 11:04:36 find no extra config, skip it, err: open /opt/aws/aws-otel-collector/etc/extracfg.txt: no such file or directory
2021-06-13T11:04:36.311Z        info    service/application.go:277      Starting aws-otel-collector...  {"Version": "v0.10.0", "NumCPU": 2}
2021-06-13T11:04:36.311Z        info    service/application.go:185      Setting up own telemetry...
2021-06-13T11:04:36.312Z        info    service/telemetry.go:98 Serving Prometheus metrics      {"address": ":8888", "level": 0, "service.instance.id": "67a80059-dddc-4804-9c93-b95d67091128"}
2021-06-13T11:04:36.312Z        info    service/application.go:220      Loading configuration...
2021-06-13T11:04:36.314Z        info    service/application.go:236      Applying configuration...
2021-06-13T11:04:36.314Z        debug   awsutil@v0.0.0-00010101000000-000000000000/conn.go:59   Using proxy address:    {"kind": "exporter", "name": "awsxray", "proxyAddr": ""}
2021-06-13T11:04:36.314Z        debug   awsutil@v0.0.0-00010101000000-000000000000/conn.go:125  Fetch region from environment variables {"kind": "exporter", "name": "awsxray", "region": "eu-west-1"}
2021-06-13T11:04:36.314Z        debug   awsxrayexporter@v0.27.0/xray_client.go:59       Using Endpoint: %s      {"kind": "exporter", "name": "awsxray", "endpoint": "https://xray.eu-west-1.amazonaws.com"}
2021-06-13T11:04:36.314Z        info    builder/exporters_builder.go:274        Exporter was built.     {"kind": "exporter", "exporter": "awsxray"}
2021-06-13T11:04:36.314Z        info    builder/exporters_builder.go:231        Ignoring exporter as it is not used by any pipeline     {"kind": "exporter"}
2021-06-13T11:04:36.314Z        info    builder/exporters_builder.go:274        Exporter was built.     {"kind": "exporter", "exporter": "awsprometheusremotewrite"}
2021-06-13T11:04:36.315Z        info    builder/exporters_builder.go:274        Exporter was built.     {"kind": "exporter", "exporter": "logging"}
2021-06-13T11:04:36.315Z        info    builder/pipelines_builder.go:204        Pipeline was built.     {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
2021-06-13T11:04:36.315Z        info    builder/pipelines_builder.go:204        Pipeline was built.     {"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-06-13T11:04:36.315Z        info    builder/receivers_builder.go:230        Receiver was built.     {"kind": "receiver", "name": "otlp", "datatype": "traces"}
2021-06-13T11:04:36.315Z        info    awsxrayreceiver@v0.27.0/receiver.go:59  Going to listen on endpoint for X-Ray segments  {"kind": "receiver", "name": "awsxray", "udp": "0.0.0.0:2000"}
2021-06-13T11:04:36.315Z        info    udppoller/poller.go:106 Listening on endpoint for X-Ray segments        {"kind": "receiver", "name": "awsxray", "udp": "0.0.0.0:2000"}
2021-06-13T11:04:36.315Z        info    awsxrayreceiver@v0.27.0/receiver.go:71  Listening on endpoint for X-Ray segments        {"kind": "receiver", "name": "awsxray", "udp": "0.0.0.0:2000"}
2021-06-13T11:04:36.315Z        debug   proxy/conn.go:98        Fetched region from environment variables       {"kind": "receiver", "name": "awsxray", "region": "eu-west-1"}
2021-06-13T11:04:36.315Z        info    builder/receivers_builder.go:230        Receiver was built.     {"kind": "receiver", "name": "awsxray", "datatype": "traces"}
2021-06-13T11:04:36.315Z        info    builder/receivers_builder.go:230        Receiver was built.     {"kind": "receiver", "name": "prometheus", "datatype": "metrics"}
2021-06-13T11:04:36.315Z        info    service/service.go:155  Starting extensions...
2021-06-13T11:04:36.315Z        info    builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "pprof"}
2021-06-13T11:04:36.316Z        info    pprofextension/pprofextension.go:79     Starting net/http/pprof server  {"kind": "extension", "name": "pprof", "config": {"TCPAddr":{"Endpoint":"localhost:1777"},"BlockProfileFraction":0,"MutexProfileFraction":0,"SaveToFile":""}}
2021-06-13T11:04:36.317Z        info    builder/extensions_builder.go:59        Extension started.      {"kind": "extension", "name": "pprof"}
2021-06-13T11:04:36.317Z        info    builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "zpages"}
2021-06-13T11:04:36.317Z        info    zpagesextension/zpagesextension.go:42   Register Host's zPages  {"kind": "extension", "name": "zpages"}
2021-06-13T11:04:36.317Z        info    zpagesextension/zpagesextension.go:55   Starting zPages extension       {"kind": "extension", "name": "zpages", "config": {"TCPAddr":{"Endpoint":"localhost:55679"}}}
2021-06-13T11:04:36.317Z        info    builder/extensions_builder.go:59        Extension started.      {"kind": "extension", "name": "zpages"}
2021-06-13T11:04:36.317Z        info    builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "health_check"}
2021-06-13T11:04:36.317Z        info    healthcheckextension/healthcheckextension.go:41 Starting health_check extension {"kind": "extension", "name": "health_check", "config": {"Port":0,"TCPAddr":{"Endpoint":"0.0.0.0:13133"}}}
2021-06-13T11:04:36.317Z        info    builder/extensions_builder.go:59        Extension started.      {"kind": "extension", "name": "health_check"}
2021-06-13T11:04:36.317Z        info    service/service.go:200  Starting exporters...
2021-06-13T11:04:36.317Z        info    builder/exporters_builder.go:92 Exporter is starting... {"kind": "exporter", "name": "awsprometheusremotewrite"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:97 Exporter started.       {"kind": "exporter", "name": "awsprometheusremotewrite"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:92 Exporter is starting... {"kind": "exporter", "name": "logging"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:97 Exporter started.       {"kind": "exporter", "name": "logging"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:92 Exporter is starting... {"kind": "exporter", "name": "awsxray"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:97 Exporter started.       {"kind": "exporter", "name": "awsxray"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:92 Exporter is starting... {"kind": "exporter", "name": "awsemf"}
2021-06-13T11:04:36.318Z        info    builder/exporters_builder.go:97 Exporter started.       {"kind": "exporter", "name": "awsemf"}
2021-06-13T11:04:36.318Z        info    service/service.go:205  Starting processors...
2021-06-13T11:04:36.318Z        info    builder/pipelines_builder.go:51 Pipeline is starting... {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
2021-06-13T11:04:36.318Z        info    builder/pipelines_builder.go:62 Pipeline is started.    {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
2021-06-13T11:04:36.318Z        info    builder/pipelines_builder.go:51 Pipeline is starting... {"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-06-13T11:04:36.318Z        info    builder/pipelines_builder.go:62 Pipeline is started.    {"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-06-13T11:04:36.318Z        info    service/service.go:210  Starting receivers...
2021-06-13T11:04:36.318Z        info    builder/receivers_builder.go:70 Receiver is starting... {"kind": "receiver", "name": "awsxray"}
2021-06-13T11:04:36.318Z        info    awsxrayreceiver@v0.27.0/receiver.go:94  X-Ray TCP proxy server started  {"kind": "receiver", "name": "awsxray"}
2021-06-13T11:04:36.318Z        info    builder/receivers_builder.go:75 Receiver started.       {"kind": "receiver", "name": "awsxray"}
2021-06-13T11:04:36.318Z        info    builder/receivers_builder.go:70 Receiver is starting... {"kind": "receiver", "name": "prometheus"}
2021-06-13T11:04:36.318Z        info    kubernetes/kubernetes.go:266    Using pod service account via in-cluster config {"kind": "receiver", "name": "prometheus", "discovery": "kubernetes"}
2021-06-13T11:04:36.319Z        debug   discovery/manager.go:195        Starting provider       {"kind": "receiver", "name": "prometheus", "provider": "kubernetes/0", "subs": "[frontend]"}
2021-06-13T11:04:36.321Z        info    builder/receivers_builder.go:75 Receiver started.       {"kind": "receiver", "name": "prometheus"}
2021-06-13T11:04:36.321Z        info    builder/receivers_builder.go:70 Receiver is starting... {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    otlpreceiver/otlp.go:83 Starting GRPC server on endpoint 0.0.0.0:4317   {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    otlpreceiver/otlp.go:145        Setting up a second GRPC listener on legacy endpoint 0.0.0.0:55680      {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    otlpreceiver/otlp.go:83 Starting GRPC server on endpoint 0.0.0.0:55680  {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    otlpreceiver/otlp.go:101        Starting HTTP server on endpoint 0.0.0.0:55681  {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    builder/receivers_builder.go:75 Receiver started.       {"kind": "receiver", "name": "otlp"}
2021-06-13T11:04:36.321Z        info    healthcheck/handler.go:128      Health Check state change       {"kind": "extension", "name": "health_check", "status": "ready"}
2021-06-13T11:04:36.321Z        info    service/application.go:197      Everything is ready. Begin running and processing data.
2021-06-13T11:04:50.257Z        INFO    loggingexporter/logging_exporter.go:57  MetricsExporter {"#metrics": 40}
2021-06-13T11:04:50.258Z        DEBUG   loggingexporter/logging_exporter.go:63  ResourceMetrics #0
Resource labels:
     -> service.name: STRING(frontend)
     -> host.name: STRING(192.168.103.224)
     -> instance: STRING(192.168.103.224:8765)
     -> port: STRING(8765)
     -> scheme: STRING(http)
     -> job: STRING(frontend)
InstrumentationLibraryMetrics #0
InstrumentationLibrary
Metric #0
Descriptor:
     -> Name: go_gc_duration_seconds
     -> Description: A summary of the pause duration of garbage collection cycles.
     -> Unit: s
     -> DataType: Summary
SummaryDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
StartTimestamp: 2021-06-13 11:04:49.849 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Count: 1596
Sum: 0.100330
QuantileValue #0: Quantile 0.000000, Value 0.000033
QuantileValue #1: Quantile 0.250000, Value 0.000047
QuantileValue #2: Quantile 0.500000, Value 0.000064
QuantileValue #3: Quantile 0.750000, Value 0.000085
QuantileValue #4: Quantile 1.000000, Value 0.000210
Metric #1
Descriptor:
     -> Name: go_goroutines
     -> Description: Number of goroutines that currently exist.
     -> Unit:
     -> DataType: DoubleGauge
DoubleDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Value: 13.000000
Metric #2
Descriptor:
     -> Name: go_info
     -> Description: Information about the Go environment.
     -> Unit:
     -> DataType: DoubleGauge
DoubleDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
     -> version: go1.16.3
StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Value: 1.000000
Metric #3
Descriptor:
     -> Name: go_memstats_alloc_bytes
     -> Description: Number of bytes allocated and still in use.
     -> Unit: By
     -> DataType: DoubleGauge
DoubleDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Value: 2436944.000000
Metric #4
Descriptor:
     -> Name:
     -> Description:
     -> Unit:
     -> DataType: None
Metric #5
Descriptor:
     -> Name: go_memstats_buck_hash_sys_bytes
     -> Description: Number of bytes used by the profiling bucket hash table.
     -> Unit: By
     -> DataType: DoubleGauge
DoubleDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls

...

StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Value: 734846976.000000
Metric #39
Descriptor:
     -> Name: process_virtual_memory_max_bytes
     -> Description: Maximum amount of virtual memory available in bytes.
     -> Unit: By
     -> DataType: DoubleGauge
DoubleDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2021-06-13 11:04:49.849 +0000 UTC
Value: 18446744073709551616.000000
2021-06-13T11:04:51.262Z        error   exporterhelper/queued_retry.go:279      Exporting failed. The error is not retryable. Dropping data.    {"kind": "exporter", "name": "awsprometheusremotewrite", "error": "Permanent error: [Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination]", "dropped_items": 35}
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send
        go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:279
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send
        go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/metrics.go:119
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1
        go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:152
github.com/jaegertracing/jaeger/pkg/queue.ConsumerFunc.Consume
        github.com/jaegertracing/jaeger@v1.22.0/pkg/queue/bounded_queue.go:104
github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumersWithFactory.func1
        github.com/jaegertracing/jaeger@v1.22.0/pkg/queue/bounded_queue.go:83
2021-06-13T11:05:04.852Z        debug   awsxrayexporter@v0.27.0/awsxray.go:54   TracesExporter  {"kind": "exporter", "name": "awsxray", "type": "awsxray", "name": "awsxray", "#spans": 1}
2021-06-13T11:05:04.854Z        debug   awsxrayexporter@v0.27.0/awsxray.go:78   request: {
  TraceSegmentDocuments: ["{\"name\":\"frontend\",\"id\":\"7c8f910d80a2a6d3\",\"start_time\":1623582304.850799,\"trace_id\":\"1-60c5e660-8c923df16597160f1c177bb3\",\"end_time\":1623582304.851821,\"http\":{\"request\":{\"method\":\"GET\",\"url\":\"http://192.168.103.224:8765/metrics\",\"user_agent\":\"Prometheus/\",\"client_ip\":\"192.168.119.57\"},\"response\":{\"status\":200,\"content_length\":0}},\"fault\":false,\"error\":false,\"aws\":{\"xray\":{\"sdk\":\"opentelemetry for go\",\"sdk_version\":\"0.18.0\",\"auto_instrumentation\":false}},\"metadata\":{\"default\":{\"http.flavor\":\"1.1\",\"http.route\":\"/metrics\",\"net.host.ip\":\"192.168.103.224\",\"net.host.port\":\"\",\"net.transport\":\"IP.TCP\",\"otel.resource.host.name\":\"frontend-5b9889c454-czgls\",\"otel.resource.service.name\":\"frontend\",\"otel.resource.telemetry.sdk.language\":\"go\",\"otel.resource.telemetry.sdk.name\":\"opentelemetry\",\"otel.resource.telemetry.sdk.version\":\"0.18.0\"}}}\n"]
}       {"kind": "exporter", "name": "awsxray"}
2021-06-13T11:05:04.963Z        debug   awsxrayexporter@v0.27.0/awsxray.go:85   response: {
  UnprocessedTraceSegments: []
}       {"kind": "exporter", "name": "awsxray"}
2021-06-13T11:05:04.965Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        info    internal/metrics_adjuster.go:302        Adjust - skipping unexpected point      {"kind": "receiver", "name": "prometheus", "type": "UNSPECIFIED"}
2021-06-13T11:05:04.966Z        INFO    loggingexporter/logging_exporter.go:57  MetricsExporter {"#metrics": 40}
2021-06-13T11:05:04.966Z        DEBUG   loggingexporter/logging_exporter.go:63  ResourceMetrics #0
Resource labels:
     -> service.name: STRING(frontend)
     -> host.name: STRING(192.168.103.224)
     -> job: STRING(frontend)
     -> instance: STRING(192.168.103.224:8765)
     -> port: STRING(8765)
     -> scheme: STRING(http)
InstrumentationLibraryMetrics #0
InstrumentationLibrary
Metric #0
Descriptor:
     -> Name: go_gc_duration_seconds
     -> Description: A summary of the pause duration of garbage collection cycles.
     -> Unit: s
     -> DataType: Summary
SummaryDataPoints #0
Data point labels:
     -> kubernetes_namespace: ho11y
     -> kubernetes_pod_name: frontend-5b9889c454-czgls
```

**Comment by pingleig** [comment]
The error comes from validation in prometheus remote write

```
2021-06-13T11:04:51.262Z        error   exporterhelper/queued_retry.go:279      Exporting failed. The error is not retryable. Dropping data.    {"kind": "exporter", "name": "awsprometheusremotewrite", "error": "Permanent error: [Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination; Permanent error: invalid temporality and type combination]", "dropped_items": 35}
```

https://github.com/open-telemetry/opentelemetry-collector/blob/3050c8efccf35937ac535cde3e8355fa8de52112/exporter/prometheusremotewriteexporter/exporter.go#L127

```go
func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) error {
					// check for valid type and temporality combination and for matching data field and type
					if ok := validateMetrics(metric); !ok {
						dropped++
						errs = append(errs, consumererror.Permanent(errors.New("invalid temporality and type combination")))
						continue
					}
}
```

I guess the is some change on prometheus receiver side, can you try build the image from latest source or use otelcontrib?
If otelcontrib works, then it's a dependency version problem, otherwise we need to see what's wrong in the prom receiver/exporter.

**Comment by mhausenblas** [comment]
Thanks a lot @pingleig and will give it a try! Could the fact that I'm exposing OM be a reason?

**Comment by mhausenblas** [comment]
Update: I can see the ADOT collector metrics in AMP but not my own (tho I've disabled OpenMetrics exposition).

**Comment by pingleig** [comment]
You mean you can see metrics from collector itself but no metrics from your application? I am not very familiar with open metrics format, but it seems it has `Metric Family` for declaring `TYPE` and is a bit different from existing prometheus.  Maybe @alolita have more context on open metrics format support in prometheus receiver.

**Comment by mhausenblas** [comment]
> You mean you can see metrics from collector itself but no metrics from your application? 

Correct, but also: I don't seem to get my app metrics even when I disable the OM exposition, so I suppose I can't blame OM?

**Comment by pingleig** [comment]
What library are you using in your go application? otel go sdk https://github.com/aws-observability/aws-otel-go or just https://github.com/prometheus/client_golang . I think prometheus is working fine for [our eks integration test](https://github.com/aws-observability/aws-otel-test-framework/tree/terraform/terraform/eks) prometheus receiver -> cloudwatch emf exporter (not AMP). (though the github action failed to [delete resource in the end](https://github.com/aws-observability/aws-otel-collector/runs/2816954319?check_suite_focus=true#logs) ...

**Comment by mhausenblas** [comment]
> What library are you using in your go application?

I'm using `github.com/prometheus/client_golang`

FWIW, I tried to build the collector but after some 10min it fails to build.

**Comment by mhausenblas** [comment]
@alolita @pingleig I've just tested with v0.11 and it *seems* this is addressed now, waiting for independent confirmation and then we can close this issue.

**Comment by alolita** [comment]
Thanks for the update @mhausenblas please let us know.
